Dockerfile

```diff
@@ -5,7 +5,11 @@ ARG AWS_SESSION_TOKEN
 
 RUN /usr/bin/aws s3 cp s3://keboola-drivers/db2-odbc/v11.5.4_linuxx64_dsdriver.tar.gz /tmp/dsdriver.tar.gz
 
-FROM php:8-cli
+# Note 2023-02-03:
+# The ODBC driver doesn't work properly with PHP >= 8.2.
+# This is probably due to changes in escaping, see: https://github.com/php/php-src/issues/8300 (PHP 8.2).
+# A password that contains "@" does not work, even if it is correct.
+FROM php:8.1-cli
```

Comment:
Na produkcii to neslo :cry: 

Funguje to s `8.1`, nefunguje s `8.2`.

Tipujem to na toto:
- https://www.php.net/ChangeLog-8.php
- https://github.com/php/php-src/issues/8300
- 
![image](https://user-images.githubusercontent.com/19371734/216715201-6a907e94-37ce-405a-877f-c55fc05212a5.png)

Ak je v hesle `@`, tak validne heslo s `8.2` neprejde, asi sa driver spolieha, ze to nebude escapovane, alebo take nieco.

-------------

Stacilo by teda v `main` branch upravit base tag.
No tato verzia s ODBC adapterom ma fixnute aj dalsie drobnosti co som komentoval a nema ziaden specialny kod (na rozdiel od PDO implementacie, co tu bola predtym), tj. malo by byt lepsie pouzit toto.

Reply:
Testovaci job uz ide (resp. spadne spravne na tom, ze tabulka neexistuje).

Job: https://connection.eu-central-1.keboola.com/admin/projects/3666/queue/528117588

![image](https://user-images.githubusercontent.com/19371734/216716473-067b876a-7da6-4640-8d70-
356a0597ca69.png)

:heavy_check_mark: connection works
![image](https://user-images.githubusercontent.com/19371734/216717271-3c117abd-9be2-4c0c-bdf2-9bcce488dc4e.png)

Reply:
Hi, I'm the person who developed the escaping feature for the ODBC extensions - I found this PR linked back from the original issue on PHP upstream. If there's a regression, I'd like to know, or have it filed as a bug upstream. I don't speak Czech, so I don't have much context to go on here.

Reply:
The connection is created as follows:
- https://github.com/keboola/db-extractor-db2/blob/2a9903613721708097194d3ef6cdbb99705cc503/src/Extractor/DB2OdbcConnection.php#L14-L29
- parent https://github.com/keboola/db-extractor-adapter/blob/7cfa4d27025d2874a9644ca21eda26a38723d753/src/ODBC/OdbcConnection.php#L60-L69

Reply:
I see, there is no `@` quoting
https://github.com/php/php-src/blob/2920a266364aef84df82cbb2b01b573705fa57ec/main/php_odbc_utils.c#L93

Maybe is the DB2 driver incompatible with the quoting format?
But why would it work for us in CI tests and not for the client? I do not know.

Reply:
We use the Db2 for i driver and haven't had any issues (in fact, the quoting thing was to solve issues of using i.e. `{` and `}` within a password. I don't know about the Db2 LUW driver.

A `@` would [result in it being quoted](https://github.com/php/php-src/blob/master/main/php_odbc_utils.c#L72); I don't think there's any other special handling for that character?

If you manually quote the password (i.e. `{p@ssword}`), does that work?

Reply:
Unfortunately, I no longer have access to the client database.

Unfortunately, I no longer have access to the client database.

I can try to replicate it against our test DB with `@` in the password, but I can't promise when I'll have time, probably no sooner than after 14 days.

Reply:
I'll try to reproduce with other drivers in the meantime then.